CODE.md

```diff
@@ -0,0 +1,23 @@
+# Coding Patterns
+
+This is here to document the design patterns chosen by the developers. It documents structures, abstractions, and philosophy in this repo.
+
+## Glossary of Terms
+
+In no particular order.
+
+_Domain_
+
+A named concept relative to the Zui app. Examples of domains are pools, queries, panes, configurations, plugins. First-class things that people can point to when they describe the app.
+
+_Main Object_
+
+The main object contains methods for managing the application in the main process. It can manage windows, session data, plugins, and app lifecycle.
+
+_Operation_
+
+A function that runs in the main process and has the main object in scope. When the app boots, operations begin listening for IPC invocations from a renderer processes. They can also be called directly from the main process using their .run() method. When invoking an operation from the main process, use the invoke function found in src/core/invoke.ts.
```

Comment:
Grammatically, this should be either "renderer processes" (without the leading "a") or "renderer process". Not sure which was intended.